### PR TITLE
[WIP] UI Polish

### PR DESCRIPTION
**Pull request by Westsaid**
Improvements to:
- Chat
TODO:
Scroll to bottom when closing chat
If scrolled, make an indication that a new message is available
- Inspect menu
- Further small changes

I will add screenshots with differences later on when this is ready.

### CODE REVIEW

File: code/ui/generalhud/inspectmenu/InspectMenu.cs [hunk 1]
```diff
@@ -174,31 +179,31 @@ public void SetConfirmationData(ConfirmationData confirmationData)
...
                     _distance?.Delete(true);
                     _suicide?.Delete(true);
 
                     if (!confirmationData.Suicide)
                     {
-                        _distance = new InspectItem(this);
+                        _distance = new InspectItem(this._itemWrapper);
                         _distance.ImageWrapper.Image.SetTexture($"/ui/inspectmenu/distance.png");
-                        _distance.InspectItemLabel.Text = $"From {confirmationData.Distance:n0}m away";
+                        _distance.InspectItemLabel.Text = $"{confirmationData.Distance:n0}m distance";
```
Comment (AMastryukov) [inline code/ui/generalhud/inspectmenu/InspectMenu.cs:202]: "Xm distance" does not sufficiently convey the idea that the number displayed is the distance from which they were killed. I don't think there is any need to change it.

Comment (Westsaid) [reply code/ui/generalhud/inspectmenu/InspectMenu.cs:202]: This is WIP so unfinishef, I have to add a better design for this to show more information on a smaller space as the current design here is 💩.

File: code/ui/generalhud/chat/ChatBox.cs [hunk 2]
```diff
@@ -126,48 +129,63 @@ private void Submit()
...
+            _canvas.AddChild(serverEntry);
+        }
         public void AddEntry(string name, string message, string avatar, LifeState lifeState, string team = null)
         {
             _lastChatFocus = 0f;
 
-            if (Messages.Count > MAX_MESSAGES_COUNT)
-            {
-                ChatEntry entry = Messages[0];
+            // if(string.IsNullOrEmpty(name)) {
```
Comment (AMastryukov) [inline code/ui/generalhud/chat/ChatBox.cs:143]: Commented out code without any explanation is probably not needed.
You can delete it or add a comment explaining why it needs to stay in.

File: code/ui/generalhud/chat/ChatBox.cs [hunk 3]
```diff
@@ -126,48 +129,63 @@ private void Submit()
...
 
-            if (!string.IsNullOrEmpty(team))
-            {
-                chatEntry.Style.BorderLeftWidth = Length.Pixels(4f);
-                chatEntry.Style.BorderLeftColor = TeamFunctions.GetTeam(team).Color;
-                chatEntry.Style.Dirty();
-            }
+            chatGroup.AddMessage(message, team);
+
+            // chatEntry.SetClass("noavatar", string.IsNullOrEmpty(avatar));
```
Comment (AMastryukov) [inline code/ui/generalhud/chat/ChatBox.cs:178]: Same as https://github.com/TTTReborn/tttreborn/pull/162/files#r687292531

File: code/ui/generalhud/chat/ChatEntry.cs [hunk 4]
```diff
@@ -1,25 +1,82 @@
...
-            NameLabel = HeadHolder.Add.Label("", "name");
-            Message = Add.Label("Message", "message");
+            // HeadHolder = Add.Panel("head");
+            // Avatar = HeadHolder.Add.Image();
+            // NameLabel = HeadHolder.Add.Label("", "name");
+            Message = Add.Label(text, "message");
+            IsNew = true;
         }
+
+        // public override void Tick()
```
Comment (AMastryukov) [inline code/ui/generalhud/chat/ChatEntry.cs:71]: https://github.com/TTTReborn/tttreborn/pull/162/files#r687292531

**Pull request closed without merging**